src/components/RichTextEditor/BlockStyleButton.tsx

```diff
@@ -17,30 +19,26 @@ export const BlockStyleButton: FC<BlockStyleButtonProps> = ({ blockType, childre
         return null;
     }
 
-    const [{ context }, send] = useActor(machineRef);
-    const { editorState } = context;
+    const editor = useSlateStatic();
+    const [blockTypeIsActive] = Editor.nodes(editor, {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        match: (n) => Element.isElement(n) && n.type === blockType,
+    });
 
-    const blockTypeIsActive = useMemo((): boolean => {
-        if (!editorState) {
-            return false;
-        }
-
-        const type = editorState.getCurrentContent().getBlockForKey(editorState.getSelection().getStartKey()).getType();
-        return type === blockType;
-    }, [editorState, blockType]);
+    const [, send] = useActor(machineRef);
```

Comment:
I would avoid `[, element]` as it can be easily miss understood and not a lot of people is aware of that writing (no longer than a week ago in PP for a PoC I had to explain it).

```suggestion
    const [_myVariable, send] = useActor(machineRef);
```
And I would prefix it with `_` as it is unused.

Reply:
i see your point, but this will raise a typescript error (also with `_`). disabling the error won't improve the dev experience IMHO